Converts notebook plots to pure Plotly

BradyPlanden opened the pull request:
# Description

This PR removes the svg based plotting methods for jupyter notebooks, changing to standard to the `plotly.show()` method. As this remove the rendered figures in GitHub, I've added a README.md to the examples directory which points users to the nbviewer rendered notebooks, which is the recommend method for plotly.

As a result of the above, this PR removes kaleido as a dependency.

## Issue reference
Fixes #410 

## Review
Before you mark your PR as ready for review, please ensure that you've considered the following:
- Updated the [CHANGELOG.md](https://github.com/pybop-team/PyBOP/blob/develop/CHANGELOG.md) in reverse chronological order (newest at the top) with a concise description of the changes, including the PR number.
- Noted any breaking changes, including details on how it might impact existing functionality.

## Type of change
- [ ] New Feature: A non-breaking change that adds new functionality.
- [ ] Optimization: A code change that improves performance.
- [ ] Examples: A change to existing or additional examples.
- [X] Bug Fix: A non-breaking change that addresses an issue.
- [ ] Documentation: Updates to documentation or new documentation for new features.
- [ ] Refactoring: Non-functional changes that improve the codebase.
- [ ] Style: Non-functional changes related to code style (formatting, naming, etc).
- [ ] Testing: Additional tests to improve coverage or confirm functionality.
- [ ] Other: (Insert description of change)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybop-team/PyBOP/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All unit tests pass: `$ nox -s tests`
- [ ] The documentation builds: `$ nox -s doctest`

You can run integration tests, unit tests, and doctests together at once, using `$ nox -s quick`.

## Further checks:
- [ ] Code is well-commented, especially in complex or unclear areas.
- [ ] Added tests that prove my fix is effective or that my feature works.
- [ ] Checked that coverage remains or improves, and added tests if necessary to maintain or increase coverage.

Thank you for contributing to our project! Your efforts help us to deliver great software.

brosaplanella reviewed (commented):
Hi Brady! I am not sure I follow what this PR changes (mostly because I am not too familiar with plotly). It seems the main change is to use a log scale in the plots, but there are still some issues with the exponents (see https://github.com/pybop-team/PyBOP/blob/7bae73c60da37932851f8373cc567f8f35e8a078/examples/notebooks/multi_model_identification.ipynb)

BradyPlanden commented:
I've now refactored this PR quite a bit, after hitting a few blockers with Kaleido and the GitHub notebook renderer, I've changed direction and reverted notebook plotting to the standard `plotly.show()` implementation. This reduces the maintenance quite a bit and is much more robust. The only drawback is that the notebook figures won't be rendered on Github. To solve this, a README has been added directing users to the nbviewer implementation.

NicolaCourtier commented on examples/README.md:
```diff
@@ -0,0 +1,43 @@
+
+<div align="center">
+
+[Notebooks on nbviewer](https://nbviewer.org/github/pybop-team/PyBOP/tree/develop/examples/notebooks/)
```
Should this direct to `main`? Please also add a link in the top level Readme.

Also, it looks like the links would remain static, so perhaps we could include links at the top of each notebook so it is possible to view any specific notebook in one click from GitHub?

NicolaCourtier reviewed (commented):
Hi @BradyPlanden, when I try to view the figures on nbviewer (e.g. [1-single-pulse-circuit-model](https://nbviewer.org/github/pybop-team/PyBOP/blob/main/examples/notebooks/LG_M50_ECM/1-single-pulse-circuit-model.ipynb)), it only shows me one figure and the rest are black rectangles. I am using Firefox on Windows, same on Chrome. Will this issue be fixed by merging?

BradyPlanden commented:
Yes it will be fixed by merging, here's the link to this branch which has the fixed plots: https://nbviewer.org/github/pybop-team/PyBOP/blob/410-bug-incorrect-axis-symbol-rendering-formu-in-notebooks/examples/notebooks/equivalent_circuit_identification.ipynb

brosaplanella reviewed (approved):
Looks good to me, thanks!
